src/shogun/classifier/Perceptron.h

```diff
@@ -72,28 +60,21 @@ class CPerceptron : public CLinearMachine
 
 	protected:
 		/** registers and initializes parameters */
-			void init();
-
-		/** train classifier
-		 *
-		 * @param data training data (parameter can be avoided if distance or
-		 * kernel-based classifiers are used and distance/kernels are
-		 * initialized with train data)
-		 *
-		 * @return whether training was successful
-		 */
-		virtual bool train_machine(CFeatures* data=NULL);
+		void init();
+		/** init the model before training begins */
```

Comment:
pls only document these methods in the base class (properly for developers, with instructions that overloading this methods allows for certain things). No need to re-document them here